### PR TITLE
feat: Add 'file type restricted' system message (WEBAPP-6980)

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: src/script/entity/message/FileTypeRestrictedMessage.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ */
+
+import {t} from 'Util/LocalizerUtil';
+
+import {SuperType} from '../../message/SuperType';
+import {Message} from './Message';
+
+/**
+ * Legal hold system message
```
Comment (bennycode) [inline src/script/entity/message/FileTypeRestrictedMessage.ts]: Wrong comment?

File: test/main.test.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ window.wire.env = {
   BACKEND_REST: 'http://localhost',
   BACKEND_WS: 'wss://localhost',
   FEATURE: {
+    ALLOWED_FILE_UPLOAD_EXTENSIONS: ['*'],
```
Comment (Yserz) [inline test/main.test.js:26]: I think for tests the default should be restrictive and should be more relaxed in specific test cases with some mocking.

E.g.:
```
spyOn<{getConfig: () => TypeUtil.RecursivePartial<Configuration>}>(Config, 'getConfig').and.returnValue({
      BACKEND_NAME: 'Test',
      FEATURE: {
        ENABLE_ACCOUNT_REGISTRATION: true,
      },
    });
```

File: src/script/util/util.ts [hunk 3]
```diff
@@ -177,6 +179,35 @@ export const trimFileExtension = (filename: string): string => {
   return '';
 };
 
+export const allowedImageTypes = ['image/bmp', 'image/gif', 'image/jpeg', 'image/jpg', 'image/png', '.jpg-large'];
```
Comment (Yserz) [inline src/script/util/util.ts:182]: Mh, not a big fan of pulling this out of the config and into a util. Config seems to be the correct place for this kind of stuff.

File: test/main.test.js (hunk 2 above)
Comment (AndyLnd) [reply test/main.test.js:26]: Very good point, I will address this in a later PR.

**Pull request merged**